crates/turbopack-css/src/process.rs

```diff
@@ -0,0 +1,747 @@
+
+use anyhow::{Context, Result};
+use indexmap::IndexMap;
+use lightningcss::{
+    css_modules::{CssModuleExport, CssModuleExports, CssModuleReference, Pattern, Segment},
+    dependencies::{Dependency, DependencyOptions, ImportDependency, Location, SourceRange},
+    error::PrinterErrorKind,
+    stylesheet::{ParserOptions, PrinterOptions, StyleSheet, ToCssResult},
+    targets::{Features, Targets},
+    values::url::Url,
+};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use smallvec::smallvec;
+use swc_core::{
+    atoms::Atom,
+    base::sourcemap::SourceMapBuilder,
+    common::{BytePos, FileName, LineCol},
+    css::{
+        ast::UrlValue,
+        codegen::{writer::basic::BasicCssWriter, CodeGenerator},
+        modules::{CssClassName, TransformConfig},
+        visit::{VisitMut, VisitMutWith},
+    },
+};
+use turbo_tasks::{ValueToString, Vc};
+use turbo_tasks_fs::{FileContent, FileSystemPath};
+use turbopack_core::{
+    asset::{Asset, AssetContent},
+    chunk::ChunkingContext,
+    reference::ModuleReferences,
+    resolve::origin::ResolveOrigin,
+    source::Source,
+    source_map::{GenerateSourceMap, OptionSourceMap},
+};
+use turbopack_swc_utils::emitter::IssueEmitter;
+
+use crate::{
+    lifetime_util::stylesheet_into_static,
+    parse::InlineSourcesContentConfig,
+    references::{
+        analyze_references,
+        url::{replace_url_references, resolve_url_reference, UrlAssetReference},
+    },
+    CssModuleAssetType,
+};
+
+// Capture up until the first "."
+static BASENAME_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[^.]*").unwrap());
+
+#[derive(Debug)]
+pub enum StyleSheetLike<'i, 'o> {
+    LightningCss(StyleSheet<'i, 'o>),
+    Swc {
+        stylesheet: swc_core::css::ast::Stylesheet,
+        css_modules: Option<SwcCssModuleMode>,
+    },
+}
+
+#[derive(Debug, Clone)]
+pub struct SwcCssModuleMode {
+    basename: String,
+    path_hash: u32,
+}
+
+impl PartialEq for StyleSheetLike<'_, '_> {
+    fn eq(&self, _: &Self) -> bool {
+        false
+    }
+}
+
+pub type CssOutput = (ToCssResult, Option<ParseCssResultSourceMap>);
+
+impl<'i, 'o> StyleSheetLike<'i, 'o> {
+    pub fn to_static(
+        &self,
+        options: ParserOptions<'static, 'static>,
+    ) -> StyleSheetLike<'static, 'static> {
+        match self {
+            StyleSheetLike::LightningCss(ss) => {
+                StyleSheetLike::LightningCss(stylesheet_into_static(ss, options))
+            }
+            StyleSheetLike::Swc {
+                stylesheet,
+                css_modules,
+            } => StyleSheetLike::Swc {
+                stylesheet: stylesheet.clone(),
+                css_modules: css_modules.clone(),
+            },
+        }
+    }
+
+    pub fn to_css(
+        &self,
+        cm: Arc<swc_core::common::SourceMap>,
+        enable_srcmap: bool,
+        remove_imports: bool,
+        handle_nesting: bool,
+    ) -> Result<CssOutput, lightningcss::error::Error<PrinterErrorKind>> {
```

Comment:
```suggestion
    ) -> Result<CssOutput> {
```

Why isn't that a anyhow::Error?